README.md

```diff
@@ -127,3 +127,12 @@ When using Magit 2.4 or newer, add this to your init script:
 (add-hook 'magit-pre-refresh-hook 'diff-hl-magit-pre-refresh)
 (add-hook 'magit-post-refresh-hook 'diff-hl-magit-post-refresh)
 ```
+
+Tramp
+-----
+
+`diff-hl` should just work with tramp. But sometimes a performance
+penalty may be experimented in slow or high latency connections. If
+you experiment such issue then an option may be to set
+`diff-hl-disable-on-remote` to `t` this will inhibit some `diff-hl`

```

Comment:
It will disable (almost?) all the features, would it not?

Reply:
`diff-hl-dired` and the others maybe??

Reply:
`diff-hl-dired` is also disabled, right?

So are there any features that keep working in remote buffers when this variable is non-nil?

Reply:
No, I didn't touch `diff-hl-dired` at all. to disable it on remoted there is `diff-hl-dired-mode-unless-remote`.

I didn't change that because I was not sure how do you prefered to proceed. 
1) Modify diff-hl-dired-mode-unless-remote
2) Create a new function something like diff-hl-dired-mode-turn-on 
```
(defun diff-hl-dired-mode-turn-on ()
  (unless (and diff-hl-disable-on-remote
	       (file-remote-p default-directory))
    (diff-hl-dired-mode)))
```
3) Modify `diff-hl-dired-mode` directly like:

```
(define-minor-mode diff-hl-dired-mode
  "Toggle VC diff highlighting on the side of a Dired window."
  :lighter ""
  (cond
   ((and diff-hl-disable-on-remote
	 (file-remote-p default-directory))
    (diff-hl-dired-mode -1))
   (diff-hl-dired-mode
    (diff-hl-maybe-define-bitmaps)
    (set (make-local-variable 'diff-hl-dired-process-buffer) nil)
    (add-hook 'dired-after-readin-hook 'diff-hl-dired-update nil t))
   (t
    (remove-hook 'dired-after-readin-hook 'diff-hl-dired-update t)
    (diff-hl-dired-clear))))
```
The first option is somehow the simplest, but a bit redundant. The second is almost duplicate the `diff-hl-dired-mode-unless-remote` and the third will limit the user to not add exceptions to this option (something he exceptionally may want to do selectively)

Reply:
Sorry, I mistook `diff-hl-dir-mode` for `diff-hl-dired-mode` inside the definition of `turn-on-diff-hl-mode`. That's where that question came from.

If the behavior in your patch satisfies your requirements, no need to tweak it further. Ultimately my question was about the phrasing in the description, which seems a little too vague. But I can fix it myself, no problem.